examples/src/uds/client.rs

```diff
@@ -5,21 +5,28 @@ pub mod hello_world {
 }
 
 use hello_world::{greeter_client::GreeterClient, HelloRequest};
+use std::env;
 #[cfg(unix)]
 use tokio::net::UnixStream;
 use tonic::transport::{Endpoint, Uri};
 use tower::service_fn;
+use ustr::ustr;
 
 #[cfg(unix)]
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Path needs to be a static str due to service_fn FnMut.
+    let path = match env::var("EXAMPLE_UDS_PATH") {
+        Ok(path) => ustr(path.as_ref()).as_str(),
```

Comment:
I want to say converting a path to a string should be possible without a third party crate?

Reply:
Would something like a `OnceLock<String>` work? AFAICT a `&'static str` is required. I used `ustr` because in an actual application one may have multiple copies of the same client but with different endpoints -- I think you'd end up reimplementing `ustr` in that case?

Reply:
Perhaps there is something obvious I missed, but when using a `String`, and for the reason I filed that other linked issue...

```
expected a closure that implements the `FnMut` trait, but this closure only implements `FnOnce`
required for `ServiceFn<{closure@examples/src/uds/client.rs:29:44: 29:57}>` to implement `Service<Uri>`
required for `ServiceFn<{closure@examples/src/uds/client.rs:29:44: 29:57}>` to implement `MakeConnection<Uri>`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B3%5D?3#file%3A%2F%2F%2FUsers%2Fkris%2FCode%2Fkriswuollett%2Ftonic%2Fexamples%2Fsrc%2Fuds%2Fclient.rs)
client.rs(29, 10): the requirement to implement `FnMut` derives from here
client.rs(31, 33): closure is `FnOnce` because it moves the variable `path` out of its environment
```